src/AzureAuth/Commands/Ado/CommandPat.cs

```diff
@@ -26,6 +38,21 @@ public class CommandPat
         private const string OutputOption = "--output";
         private const string OutputHelp = "How PAT information is displayed. [default: token]\n[possible values: none, status, token, base64, header, headervalue, json]";
 
+        private const string DomainOption = "--domain";
+        private const string DomainHelp = "The preferred domain used when acquiring Azure Active Directory access tokens. [default: microsoft.com]";
+
+        private const string AccessTokenPrompt = "AzureAuth ADO PAT";
+        private static readonly IEnumerable<AuthMode> AccessTokenAuthModes = new[] { AuthMode.Default };
+        private static readonly TimeSpan AccessTokenTimeout = TimeSpan.FromMinutes(15);

```

Comment:
I think when we originally discussed this feature we talked about keeping those kinds of flags to a minimum and just using the defaults (which should provide the best experience). Timeout certainly makes sense to me, but should we give a lot of control over the access token mode if we already have sane defaults?

Reply:
I think the main consideration for auth modes is headless. The default modes do not include device code since 99% of the time it won't work with how things use AzureAuth. So imagine a mac or linux ssh session wanting to get and cache a PAT. They need to be able to say `azureauth ado pat --mode devicecode`

Reply:
I suppose for that scenario, one _could_ run `azureauth ado token --mode devicecode` first, to cache AAD auth, and then could call the pat command as a workaround.

Reply:
I hesitated to add something like this in the first place because using an access token to get a PAT is an implementation detail and I didn't want to give too much control over that at the top level.

Headless usage is a good consideration though. I don't see requiring certain users to make additional calls just to warm a cache as a good option. What would you call this flag? Just `--mode` for consistency? `--access-token-mode` for specificity? 🤔

Reply:
I like the option `--mode`

Reply:
From our offline discussion, I am generally pro "don't add it until you need it" but I think keeping the controls for auth consistent across the commands that need to auth. Also because of the overhead of shipping new versions internally atm, I'd rather have it and not need it, then need it and not have it.